### PR TITLE
[RFC] Draw phylogenetic tree

### PR DESCRIPTION
**Pull request by Unia**
This PR adds a complete phylogenetic tree implementation. It also changes the main window, in that the DNA view and the phylogenetic tree view are together wrapped in a split pane.

### CODE REVIEW

File: src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	 * @return The center x-coordinate.
+	 */
+	public final double getCenterX() {
+		return centerX.get();
+	}
+
+	/**
+	 * @return The center x property.
+	 */
+	public DoubleProperty centerXProperty() {
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java]: why is every method involving this property final, except for this one?

File: src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	 * @return The center y-coordinate.
+	 */
+	public final double getCenterY() {
+		return centerY.get();
+	}
+
+	/**
+	 * @return The center y property.
+	 */
+	public DoubleProperty centerYProperty() {
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java]: again,  why is every method involving this property final, except for this one?

File: src/main/java/nl/tudelft/dnainator/ui/widgets/PhylogeneticTree.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+		// Draw self and attach children's edges to it.
+		PhylogeneticNode self = new PhylogeneticNode(x, first.getCenterY()
+				+ (last.getCenterY() - first.getCenterY()) / 2, SQUARE);
+		for (PhylogeneticNode child : drawnChildren) {
+			child.getOutgoingEdge().bindTo(self);
+		}
+		return self;
+	}
+
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/PhylogeneticTree.java]: Is this a redundant newline?

File: src/main/java/nl/tudelft/dnainator/ui/widgets/PhylogeneticTree.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	private PhylogeneticNode drawInternal(TreeNode root, List<TreeNode> children,
+			double x, double y) {
+		List<PhylogeneticNode> drawnChildren = new ArrayList<>();
+
+		// Draw all children and attach an outgoing edge.
+		for (int i = 0; i < children.size(); i++) {
+			PhylogeneticNode node = draw(children.get(i), x + LEVELWIDTH, y + (i * LEAFHEIGHT));
+			drawnChildren.add(node);
+			PhylogeneticEdge edge = new PhylogeneticEdge(node, x);
+			node.setOutgoingEdge(edge);
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/PhylogeneticTree.java]: You might consider renaming this to incoming edge.

File: src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java (hunk 1 above)
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java]: Convention, see: http://docs.oracle.com/javafx/2/binding/jfxpub-binding.htm#CHDEHBAC

File: src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java (hunk 2 above)
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java]: See above :)

File: src/main/java/nl/tudelft/dnainator/ui/widgets/PhylogeneticTree.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	private PhylogeneticNode draw(TreeNode node, double x, double y) {
+		List<TreeNode> children = node.getChildren();
+		if (children.size() == 0) {
+			return drawLeaf(node, x);
+		}
+		return drawInternal(node, children, x, y);
+	}
+
+	private PhylogeneticNode drawLeaf(TreeNode leaf, double x) {
+		double y = currentLeafY;
```
Comment (gfokkema) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/PhylogeneticTree.java]: See #78 .
The y positioning based on an internal variable doesn't feel right.

File: src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	 */
+	public void setOutgoingEdge(PhylogeneticEdge edge) {
+		this.outgoingEdge = edge;
+	}
+
+	/**
+	 * @return This {@link PhylogeneticNode}'s outgoing {@link PhylogeneticEdge}.
+	 */
+	public PhylogeneticEdge getOutgoingEdge() {
+		return this.outgoingEdge;
```
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/ui/drawables/PhylogeneticNode.java]: Remove "this" for consistency with other getters?

**Comment by gfokkema** [comment]
Looks good, merging.

**Pull request merged**